Source/Core/Core/Lua/LuaFunctions/LuaRegisters.cpp

```diff
@@ -0,0 +1,467 @@
+
+  std::array luaRegistersFunctionsWithVersionsAttached = {
+    luaL_Reg_With_Version({"getRegister", "1.0", getRegister}),
+    luaL_Reg_With_Version({"getRegisterAsUnsignedByteArray", "1.0", getRegisterAsUnsignedByteArray}),
+    luaL_Reg_With_Version({"getRegisterAsSignedByteArray", "1.0", getRegisterAsSignedByteArray}),
+    luaL_Reg_With_Version({"setRegister", "1.0", setRegister}),
+    luaL_Reg_With_Version({"setRegisterFromByteArray", "1.0", setRegisterFromByteArray})
+  };
+
+  std::unordered_map<std::string, std::string> deprecatedFunctionsMap = std::unordered_map<std::string, std::string>();
+
+  addLatestFunctionsForVersion(luaRegistersFunctionsWithVersionsAttached, luaApiVersion, deprecatedFunctionsMap, luaState);
+  lua_setglobal(luaState, "registers");
+}
+
+class RegisterObject
+{
+public:
+
+  enum REGISTER_TYPE
+  {
+    GENERAL_PURPOSE_REGISTER,
+    FLOATING_POINT_REGISTER,
+    PC_REGISTER,
+    RETURN_REGISTER, //LR register
+    UNDEFINED
+  };
+
+  RegisterObject(REGISTER_TYPE newRegType, u8 newRegNum) {
+    regType = newRegType;
+    registerNum = newRegNum;
+  }
+
+  u8 registerNum;
+  REGISTER_TYPE regType;
+};
+
+RegisterObject parseRegister(const char* registerString)
+{
+  if (registerString == NULL || registerString[0] == '\0')
+    return RegisterObject(RegisterObject::REGISTER_TYPE::UNDEFINED, 0);
+
+  s64 regNum = 0;
+  switch (registerString[0])
+  {
+  case 'r':
+  case 'R':
+
+    regNum = std::stoi(std::string(registerString).substr(1), NULL);
+    if (regNum < 0 || regNum > 31)
+      return RegisterObject(RegisterObject::REGISTER_TYPE::UNDEFINED, 0);
+    return RegisterObject(RegisterObject::REGISTER_TYPE::GENERAL_PURPOSE_REGISTER, regNum);
+    break;
+
+  case 'f':
+  case 'F':
+    regNum = std::stoi(std::string(registerString).substr(1), NULL);
+    if (regNum < 0 || regNum > 31)
+      return RegisterObject(RegisterObject::REGISTER_TYPE::UNDEFINED, 0);
+    return RegisterObject(RegisterObject::REGISTER_TYPE::FLOATING_POINT_REGISTER, regNum);
+    break;
+
+  case 'p':
+  case 'P':
+    if (registerString[1] != 'c' && registerString[1] != 'C')
+      return RegisterObject(RegisterObject::REGISTER_TYPE::UNDEFINED, 0);
+    return RegisterObject(RegisterObject::REGISTER_TYPE::PC_REGISTER, 0);
+    break;
+
+  case 'l':
+  case 'L':
+    if (registerString[1] != 'r' && registerString[1] != 'R')
+      return RegisterObject(RegisterObject::REGISTER_TYPE::UNDEFINED, 0);
+    return RegisterObject(RegisterObject::REGISTER_TYPE::RETURN_REGISTER, 0);
+    break;
+
+  default:
+    return RegisterObject(RegisterObject::REGISTER_TYPE::UNDEFINED, 0);
+  }
+}
+
+u8* getAddressForRegister(RegisterObject registerObject, lua_State* luaState, const char* funcName)
+{
+  u8 regNum = 0;
+  u8* address = NULL;
+  switch (registerObject.regType)
+  {
+  case RegisterObject::REGISTER_TYPE::GENERAL_PURPOSE_REGISTER:
+    regNum = registerObject.registerNum;
+    address = (u8*)(rGPR + regNum);
```

Comment:
please explicitly use `reinterpret_cast` for casts like these, rather than C-style casts.